docs/use_cases/otel/insights.md

```diff
@@ -0,0 +1,127 @@
+# Insights
+
+In the previous step, we preprocessed the data to enhance its usability. Next, we want to generate valuable insights from it.
+
+### P95 Latency
+
+P95 latency is a common statistical measure used for network performance analysis. It means that 95% of requests were served faster than this value, while 5% took longer than this.
+
+#### Step 1: Define a Tumbling Window
+
+To compute P95 latency over time, we first define a **10-second tumbling window** on the `spans` view.
+
+```sql
+CREATE LOCAL VIEW spans_tumble_10s AS
+SELECT * FROM TABLE(
+	TUMBLE(
+	TABLE spans,
+	DESCRIPTOR(eventTime),
+	INTERVAL '10' SECOND
+	)
+);
+```
+
+This groups spans into non-overlapping 10-second time windows.
+
+#### Step 2: Define a Rust UDF for P95 Calculation
+
+Here we leverage Feldera's support for Rust [UDFs](https://docs.feldera.com/sql/udf) to implement a function that given an array of integers, returns the 95th percentile value from it.
+
+##### SQL Definition for P95 Rust UDF
+
+```sql
+-- UDF to calculate p95 value given an integer array
+CREATE FUNCTION p95(x BIGINT ARRAY NOT NULL) RETURNS BIGINT;
+```
+
+##### Rust Implementation of P95
+
+```rust
+pub fn p95(x: Vec<Option<i64>>) -> Result<Option<i64>, Box<dyn std::error::Error>> {
```

Comment:
SQL has an PERCENTILE function, but we don't support it yet, we should. It may also be a bit more efficient.
But for now you can say that PERCENTILE is not yet implemented so you do it in Rust.
This seems to be PERCENTILE_CONT you are implementing.
(There is also a PERCENTILE_DISC)